lib/Dialect/FIRRTL/FIRRTLTypes.cpp

```diff
@@ -1533,6 +1553,32 @@ BundleType::getElementTypePreservingConst(size_t index) {
   return type.getConstType(type.isConst() || isConst());
 }
 
+/// Return this type with any type aliases recursively removed from itself.
+FIRRTLBaseType BundleType::getAnonymousType() {
+  auto *impl = getImpl();
+
+  // If we've already determined and cached the anonymous type, use it.
+  if (impl->anonymousType)
+    return impl->anonymousType;
+
+  // If this type is already anonymous, use it and remember for next time.
+  if (!impl->props.containsTypeAlias) {
```

Comment:
Not sure if it makes a difference, but this could be the first check and just return `*this` without setting the `anonymousType`, maybe.

Reply:
Yeah, that makes sense and I agree. I implemented in a consistent way to `getPassiveType` implementation so I'd like to prioritize the consistency here. I think it's better to change the style at the same time. Thank you for the suggestion!